cancer_ai/rewarder_test.py

```diff
@@ -0,0 +1,301 @@
+import unittest
+from datetime import datetime, timedelta
+from rewarder import CompetitionLeader, RewarderConfig, Rewarder
+
+class TestRewarder(unittest.TestCase):
+
+    def test_single_competition_single_leader(self):
+        """Test case 1: Only one competition with 1 leader -> leader takes it all"""
+        competitions_leaders = {
+            "competition-1": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now())
+        }
+        scores = {"leader-1": 0}
+        rewarder_config = RewarderConfig(competition_leader_mapping=competitions_leaders, scores=scores)
+        rewarder = Rewarder(config=rewarder_config)
+        
+        rewarder.update_scores()
+
+        # Assert that the leader takes it all
+        self.assertAlmostEqual(rewarder.scores["leader-1"], 1.0)
+
+    def test_three_competitions_three_leaders_no_reduction(self):
+        """Test case 2: 3 competitions with 3 different leaders, no reduction -> all have 33% of the shares"""
+        reward_split_by_three = 1 / 3
+        competitions_leaders = {
+            "competition-1": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now()),
+            "competition-2": CompetitionLeader(hotkey="leader-2", leader_since=datetime.now()),
+            "competition-3": CompetitionLeader(hotkey="leader-3", leader_since=datetime.now())
+        }
+        scores = {"leader-1": 0, "leader-2": 0, "leader-3": 0}
+        rewarder_config = RewarderConfig(competition_leader_mapping=competitions_leaders, scores=scores)
+        rewarder = Rewarder(config=rewarder_config)
+        rewarder.update_scores()
+
+        # Assert that all leaders have roughly 1/3 of the shares
+        self.assertAlmostEqual(rewarder.scores["leader-1"], reward_split_by_three, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], reward_split_by_three, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-3"], reward_split_by_three, places=2)
+
+    def test_three_competitions_three_leaders_with_reduction(self):
+        """Test case 3: 3 competitions with 3 different leaders, one has a reduced share by 10%"""
+        first_competion_leader_since = datetime.now() - timedelta(days=21)
+
+        base_share = 1/3
```

Comment:
for simple cases like this values should be hand calculated and provided on top of the method